### PR TITLE
Combine Lab build and Deploys into a single manifest that is deployable by instructors or students

### PR DESCRIPTION
**Pull request by patricksimonian**
The labs didn't have a build config and depended on running an `oc new-build` command with certain environment variables to build the labs. This has been abstracted into a buildconfig. In addition, the build and deploy were combined together to make the lab a single deployable unit. 

## But this goes against our patterns?

Yes, typically apps are built/deployed in two different namespaces. The workshops though are a bit different. They have a few requirements that make it much easier to manage when the build/deploys are combined together. 

- deployable quickly with minimal knowledge of the app
- should be deployable by the students (for 201)
> Over the last few Openshift 101 courses, we have found it incredibly useful to make lab instances for each student. This way we can apply variable substitution for links and references with their `Github IDS`.

In addition, the metadata for all objects have been unified so that deploys will wait for the builds to complete which makes for a better ux.

### CODE REVIEW

**Comment by patricksimonian** [comment]
@cvarjao will this affect the `workbench`

File: workshop-labs/docs/openshift201/01_git_setup.md [hunk 1]
```diff
@@ -26,12 +26,12 @@ git push --set-upstream origin [username]-201
...
-Navigate to the public repository (<https://github.com/{admin}/devops-platform-workshops-labs)>
+Navigate to the public repository (<https://github.com/[username]/devops-platform-workshops-labs)>
 
-- Click "Branch:" and add your [username]-201 as the branch name.
-- Click "Create branch: [username]-201"
+- Click "Branch:" and add you [username]-201 as the branch name.
+- Click "Create branch [username]-201"
 
-Follow the above instructions to "Create your branch with the CLI" but exclude the `git branch` and `git push` commands to clone a local copy and switch to the already created [username]-201 branch
+Follow the above instructions to "Create your branch with the CLI" but exclude the `git branch` and `git push` commands to clone a local copy and switch to the already create [username]-201 branch
```
Comment (ShellyXueHan) [inline workshop-labs/docs/openshift201/01_git_setup.md:34]: haha I think `created` is correct (not that im challenging your english :P)

File: workshop-labs/config/Caddyfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+:4000
+root * /opt/app-root/book-openshift201
```
[the remark this answers is not in the document]
Comment (patricksimonian) [reply workshop-labs/config/Caddyfile:2]: hmm ill add comments to this to mention changing the root depending on what workshop you want to run
this is just for runniing the server locally with `docker run`

File: workshop-labs/docs/openshift201/01_git_setup.md (hunk 1 above)
Comment (patricksimonian) [reply workshop-labs/docs/openshift201/01_git_setup.md:34]: you should be.

**Comment by jefkel** [comment]
a concern I have is that we're removing learning opportunities for the class.  The course was not meant to be consumed as a lecture + easy path lab.  The labs (and issues the attendee's have) was meant to be slightly challenging.  There is a difference between making things easier for the instructors to setup vs making things too easy for the students to consume.

Having the attendee's bumping into each other's deployments and builds allows learning on how to name your apps, how to share a space, etc.

Having the separation of the build/deploy namespaces is meant to be an introduction to the patterns we encourage on the platform.  Other than making this *easier* on the students and instructors, how does this prepare them for developing on the BC Gov platform?

**Comment by patricksimonian** [comment]
Totally, 

I think to clarify the intention of this lab is not to combine exercises into a single namespace. This is purely just for students/instructors to host the lab git-book app its self. The lab content still exercises the use of our best practices for having multiple namespaces.

Does that help at all? As far as consuming the labs. We had always provided them a link. I was thinking more along the lines of letting them see some of the features of Openshift, UI changes when build and deploy configs are joined together. As well as make it as easy as possible for content developers/instructors to build and develop the labs :)

**Pull request merged**